src/app/app.module.ts

```diff
@@ -8,21 +9,43 @@ import { MainPageComponent } from './main-page/main-page.component';
 import { ProductPageComponent } from './product-page/product-page.component';
 import { CartPageComponent } from './cart-page/cart-page.component';
 import { HttpClientModule } from '@angular/common/http';
+import { SignUpPageComponent } from './sign-up-page/sign-up-page.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { UserLoginPageComponent } from './user-login-page/user-login-page.component';
+import { StoreModule } from '@ngrx/store';
+import { environment } from 'src/environments/environment';
+import { EffectsModule } from '@ngrx/effects';
+import { AuthCustomersService } from './shared/auth-customers.service';
+import { AuthEffects } from './shared/store/user/auth.effects';
+import { reducers } from './shared/store/app.states';
+import { AuthModule } from './shared/store/auth.module';
 
 @NgModule({
   declarations: [
     AppComponent,
     MainLayoutComponent,
     MainPageComponent,
     ProductPageComponent,
-    CartPageComponent
+    CartPageComponent,
+    SignUpPageComponent,
+    UserLoginPageComponent
   ],
   imports: [
     BrowserModule,
     AppRoutingModule,
-    HttpClientModule
+    HttpClientModule,
+    FormsModule,
+    ReactiveFormsModule,
+
+    StoreModule.forRoot(reducers),
+    EffectsModule.forRoot([AuthEffects]),
+    StoreDevtoolsModule.instrument({
+      maxAge: 25,
+      logOnly: environment.production,
+    }),
+    AuthModule, // уточнить этот вопрос
   ],
-  providers: [],
+  providers: [AuthCustomersService],
```

Comment:
Did you provide `AuthCustomerService` in `AuthModule`? If yes then you don't need provide it here (because you imported `AuthModule`)